src/tests/ftest/pool/destroy.py

```diff
@@ -498,3 +498,90 @@ def test_forcedestroy_connected(self):
+
+        :avocado: tags=all,pr,daily_regression
+        :avocado: tags=vm
+        :avocado: tags=pool,pool_destroy
+        :avocado: tags=pool_destroy_with_containers,test_destroy_with_containers
+        """
+        hostlist_servers = self.hostlist_servers[0:1]
+
+        # Start servers
+        self.start_servers(self.get_group(self.server_group, hostlist_servers))
+
+        # Create the pool
+        self.validate_pool_creation(hostlist_servers)
+
+        # Create pool container
+        self.add_container(pool=self.pool)
+
+        # Destroy pool with recursive unset
+        self.log.info("Attempting to destroy a pool with containers")
+        exception_detected = False
+        try:
+            self.pool.destroy(recursive=0, disconnect=0)
+        except TestFail as result:
+            exception_detected = True
+            self.log.info(
+                "Expected exception - destroying pool with containers: %s",
+                str(result))
+
+        if not exception_detected:
+            # The pool-destroy did not raise an exception as expected
+            self.log.error(
+                "Exception did not occur - destroying pool with containers")
+
+            # Prevent attempting to delete the pool in tearDown()
+            self.pool.pool = None
+
+        self.log.info("Check if files still exist")
+        self.assertTrue(
+            self.pool.check_files(hostlist_servers),
+            "Pool UUID {} should not be removed when containers exist".format(self.pool.uuid))
+
+        self.assertTrue(
+            exception_detected, "No exception when deleting a pool with containers")
+
+    def test_recursivedestroy_with_containers(self):
+        """Recursively destroy pool with existing containers.
+
+        Test destroying a pool that has existing containers with recursive == true.
+        Should pass.
+
+        :avocado: tags=all,pr,daily_regression
+        :avocado: tags=vm
+        :avocado: tags=pool,pool_destroy
+        :avocado: tags=pool_recursive_destroy_with_containers,test_recursivedestroy_with_containers
+        """
+        hostlist_servers = self.hostlist_servers[0:1]
+
+        # Start servers
+        self.start_servers(self.get_group(self.server_group, hostlist_servers))
+
+        # Create the pool
+        self.validate_pool_creation(hostlist_servers)
+
+        # Create pool container
+        self.add_container(pool=self.pool)
+
+        # Destroy pool with recursive set
+        self.log.info("Attempting to recursively destroy a pool with containers")
+        exception_detected = False
+        try:
+            self.pool.destroy(recursive=1, disconnect=0)
+        except TestFail as result:
+            exception_detected = True
+            self.log.info(
+                "Unexpected exception - destroying pool with containers: %s",
+                str(result))
+        finally:
+            # Prevent attempting to delete the pool in tearDown()
+            self.pool.pool = None
```

Comment:
```suggestion
            self.container = None
```
The pool was successfully destroyed, but the container object is still there and the `tearDown()` will try to destroy it and that will fail, so delete the object.

Reply:
done